### ISSUE TITLE
Support model info targets with function and a non-%value expression

### ISSUE THREAD

**Issue by msargentibm** [cqframework/clinical_quality_language#578]
I was attempting to translate cql to elm using USCore. CQL Definition:
```
library "Test" version '1.0.0'
using "USCore" version '3.1.0'
parameter MeasurementPeriod default Interval[
  @1900-01-01T00:00:00.0,
  @2020-12-31T00:00:00.0
]

context Patient

define "getPedBMI":
  [PediatricBMIforAgeObservationProfile] P
    where exists(P.VSCat)
```

The translation succeeds, but the elm that is output appears to be incomplete. In uscore-modelinfo-3.1.0.xml, VSCat makes use of FHIRHelpers with an argument that contain some complex logic:
```
  <typeInfo baseType="USCore.DomainResource" namespace="USCore" name="PediatricBMIforAgeObservationProfile" identifier="http://hl7.org/fhir/us/core/StructureDefinition/pediatric-bmi-for-age" label="US Core Pediatric BMI for Age Observation Profile" retrievable="true" primaryCodePath="code" xsi:type="ClassInfo">
 ... SNIPPED ...
      <element name="VSCat" elementType="System.Concept" target="FHIRHelpers.ToConcept(%parent.category[coding.system='http://terminology.hl7.org/CodeSystem/observation-category',coding.code='vital-signs'].value)"/>
```

However, all information on how to retrieve the VSCat is dropped in the output elm. Only the FHIRHelpers reference is maintained. Output snippet:
```
      <def name="getPedBMI" context="Patient" accessLevel="Public">
         <expression xsi:type="Query">
            <source alias="P">
               <expression xmlns:ns2="http://hl7.org/fhir/us/core" dataType="ns2:PediatricBMIforAgeObservationProfile" templateId="http://hl7.org/fhir/us/core/StructureDefinition/pediatric-bmi-for-age" xsi:type="Retrieve"/>
            </source>
            <where xsi:type="Exists">
               <operand xsi:type="ToList">
                  <operand name="ToConcept" libraryName="FHIRHelpers" xsi:type="FunctionRef">
                     <operand path="VSCat" scope="P" xsi:type="Property"/>
                  </operand>
               </operand>
            </where>
         </expression>
      </def>
```

I found the place in code where the information was being dropped. In the LibraryBuilder class, when a function on an element's target is processed there is no attempt to further process the arguments at all (I think there is an assumption that the argument is always `%value`). I have a proposed tweak to try to capture the information from the function arguments: https://github.com/cqframework/clinical_quality_language/compare/master...msargentibm:function-recursion

With this change, re-running the same cql the elm contains information about the FHIRHelpers function being used as well as the information to locate the property.
```
      <def name="getPedBMI" context="Patient" accessLevel="Public">
         <expression xsi:type="Query">
            <source alias="P">
               <expression xmlns:ns2="http://hl7.org/fhir/us/core" dataType="ns2:PediatricBMIforAgeObservationProfile" templateId="http://hl7.org/fhir/us/core/StructureDefinition/pediatric-bmi-for-age" xsi:type="Retrieve"/>
            </source>
            <where xsi:type="Exists">
               <operand xsi:type="ToList">
                  <operand name="ToConcept" libraryName="FHIRHelpers" xsi:type="FunctionRef">
                     <operand path="value" xsi:type="Property">
                        <source xsi:type="SingletonFrom">
                           <operand xsi:type="Query">
                              <source alias="$this">
                                 <expression path="category" xsi:type="Property">
                                    <source name="P" xsi:type="AliasRef"/>
                                 </expression>
                              </source>
                              <where xsi:type="And">
                                 <operand xsi:type="Equal">
                                    <operand path="system" xsi:type="Property">
                                       <source xsi:type="First">
                                          <source path="coding" scope="$this" xsi:type="Property"/>
                                       </source>
                                    </operand>
                                    <operand valueType="t:String" value="http://terminology.hl7.org/CodeSystem/observation-category" xsi:type="Literal"/>
                                 </operand>
                                 <operand xsi:type="Equal">
                                    <operand path="code" xsi:type="Property">
                                       <source xsi:type="First">
                                          <source path="coding" scope="$this" xsi:type="Property"/>
                                       </source>
                                    </operand>
                                    <operand valueType="t:String" value="vital-signs" xsi:type="Literal"/>
     SNIPPED ...
```

I'm pretty new to working with CQL, ELM, and this codebase, but I would be interested to see what folks familiar with the code think of the change. Is it a safe change to make, or does it only work for this specific use case and open up a can of worms I haven't considered yet for other cases?

Thanks!

**Comment by brynrhodes** [comment]
Hi @msargentibm , this is great! The mapping expansion is a work in progress, so we would definitely welcome contributions along these lines. Feel free to submit this as a PR and I'll happily review!

**Comment by msargentibm** [comment]
Thanks @brynrhodes 

PR is opened. I will keep an eye out so I can help address any issues or additions that are needed.

**Comment by msargentibm** [comment]
Any chance someone would be able to look at the PR for an initial round of feedback? Am I supposed to manually add reviewers to the PR?

**Comment by brynrhodes** [comment]
Apologies for the delays here, I've been heads down on other things and this fell off my radar. I've submitted a review, the change looks good, I just requested a test case.

**Comment by brynrhodes** [comment]
Addressed in the 1.5.2 release
